### DISCUSSION TITLE
How to change only the hk array in the LPF package?

### DISCUSSION THREAD

**Discussion by javgs-bd** [modflowpy/flopy#1874] [Q&A] [answered]
Hi,
I'm just wondering if there is a way of just replacing the hk array in the lpf package without reconstructing it, or by reconstructing it but passing in a error-free way all the original arguments except for the new array.

I have my model loaded as ````gwf````, and I have access to the hk array via the ````gwf.lpf.hk.array````.

Trying ````gwf.lpf.hk.array = new_hk```` results in:

````
AttributeError                            Traceback (most recent call last)
Cell In[119], line 1
----> 1 gwf.lpf.hk.array = new_hk

File ~\pyenvs\flopy\lib\site-packages\flopy\utils\util_array.py:632, in Util3d.__setattr__(self, key, value)
    629         u2d.how = value
    630 else:
    631     # set the attribute for u3d
--> 632     super().__setattr__(key, value)

AttributeError: can't set attribute 'array'
```` 

It would seems the obvious way would be to pass this new_hk array to the constructor of the lpf package, but I'm wondering if by repeating arguments I would leave out some important definition. The model is a mf-usg and the LPF package uses external files for hk, kz, and sy at each layer.

Any help would be appreciated.

**Comment by hjia1005** [comment]
For mfusg model, I think you have to set property values by layer. Below is what I would do in your case:
```
for layer in range(gwf.nlay):
    gwf.lpf.hk[layer]=new_hk[layer]
```
new_hk is a list and new_hk[layer] is the new property array for that layer.

**Reply by javgs-bd** [reply, marked as answer]
Thanks @hjia1005. Actually, I can confirm now that you can do:

````gwf.lpf.hk = new_hk````

for a mf-usg model where the new_hk is one dimensional and contains the total number of nodes in the grid.

Thanks for pointing the path to go!
